chore: remove mailbox configs from warp route deployments

mshojaei-txfusion opened the pull request:
### Description

This PR removes mailbox configurations from all warp route deployment files in the registry. This change is part of the effort to clean up and standardize deployment configurations across the Hyperlane ecosystem.

The changes:
- Remove all `mailbox:` entries from warp route deployment files
- Maintain all other configuration parameters (interchainSecurityModule, owner, type, etc.)
- No functional changes to the deployment process

### Backward compatibility

Yes. This change is backward compatible as it only removes redundant configuration fields that are not used in the deployment process. The removal of mailbox configurations does not affect the functionality of the warp routes or their deployment process.

### Testing

 The changes have been verified by:
- Verifying the YAML structure remains valid
- Confirming all other deployment parameters remain unchanged

Resolves https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/5258
Resolves https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/5374
Fixes ENG-1057

ltyu commented:
make sure to pull any new configs and also remove mailbox

ltyu commented:
as noted on slack. do not merge until this is complete: https://linear.app/hyperlane-xyz/issue/ENG-1018/remove-mailbox-from-warp-config-in-infra-registry

mshojaei-txfusion commented:
> as noted on slack. do not merge until this is complete: https://linear.app/hyperlane-xyz/issue/ENG-1018/remove-mailbox-from-warp-config-in-infra-registry

So I will pull and remove once we had it ready.

antigremlin commented:
We need to add a few more files:
- deployments/warp_routes/CBBTC/base-ethereum-superseed-deploy.yaml
- deployments/warp_routes/UBTC/boba-bsquared-soneium-swell-deploy.yaml
- deployments/warp_routes/USDC/arbitrum-base-ethereum-ink-optimism-solanamainnet-superseed-deploy.yaml

mshojaei-txfusion commented:
> We need to add a few more files:
> 
> * deployments/warp_routes/CBBTC/base-ethereum-superseed-deploy.yaml
> * deployments/warp_routes/UBTC/boba-bsquared-soneium-swell-deploy.yaml
> * deployments/warp_routes/USDC/arbitrum-base-ethereum-ink-optimism-solanamainnet-superseed-deploy.yaml

Will do final removal after Le removed them from Infra, so we do it once on latest version

mshojaei-txfusion commented:
@antigremlin Now that you merged Infra, I updated this branch with main and removed all mailbox addresses.
If all good we can merge this.
CC: @nambrot @ltyu
